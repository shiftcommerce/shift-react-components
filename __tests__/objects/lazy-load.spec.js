// Libraries
import React from 'react'

// Objects
import LazyLoad from '../../src/objects/lazy-load'

test('renders a placeholder when image is loading', () => {
  // act
  const wrapper = mount(
    <LazyLoad height='35px' width='125px' src='test@test.com' />
  )

  // assert
  expect(wrapper).toMatchSnapshot()

  // Assert lazyload component is loading image
  expect(wrapper.find('.lazyload-placeholder')).toHaveClassName('lazyload-placeholder')
})
