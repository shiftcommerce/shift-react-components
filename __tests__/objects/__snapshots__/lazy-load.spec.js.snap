// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a placeholder when image is loading 1`] = `
<LazyLoad
  height="35px"
  src="test@test.com"
  width="125px"
>
  <div
    style={
      Object {
        "backgroundColor": "#EEEFF3",
        "height": 0,
        "paddingBottom": "NaN%",
        "width": "100%",
      }
    }
  >
    <FadeIn
      offset={150}
      once={true}
      render={[Function]}
    >
      <LazyLoad
        offset={150}
        once={true}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <div
          className="lazyload-placeholder"
          style={
            Object {
              "height": undefined,
            }
          }
        />
      </LazyLoad>
    </FadeIn>
  </div>
</LazyLoad>
`;
